### PR TITLE
feat: CB Grafana Dashboard and PR comments

### PR DESCRIPTION
**Pull request by JuanPedroGHM**
## Due Diligence

- General:
    - [x]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [x] unit tests: all split configurations tested
    - [x] unit tests: multiple dtypes tested
    - [x] documentation updated where needed

## Description

Small changes to bench_trigger.yml to remove those ugly "x" on the main page.

Extra changes:

- Benchmark results get posted as a comment on the PR conversation. Two type of changes are posted:
  - A difference of ten percent compared to the last value on the database (either positive or negative change).
  - When a value is in not within 90 percent of the measured values in the last 90 days (Lower than the 5% percentile, higher than the 95% percentile)
  - Values can be adjusted on the [gitlab cb repostiory](https://codebase.helmholtz.cloud/helmholtz-analytics/cb)
- Some alerts are to be expected, as the benchmarks are running in the Horeka accelerated nodes. Everything got a bit faster and a lot more stable.
- The cb/report status does not reflect the results of the benchmarks, only if the pipeline ran successfully or not. 
- Added links to the Grafana dashboard. Access should work without login data. [Link here](https://57bc8d92-72f2-4869-accd-435ec06365cb.ka.bw-cloud-instance.org:3000/d/adjpqduq9r7k0a/heat-cb?orgId=1)
- Values will not be saved in the old system any more, bench_report.yaml action is disabled.

Issue/s resolved: #

## Type of change
- Bug fix (non-breaking change which fixes an issue)

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by JuanPedroGHM** [comment]
### Benchmarks results - Sponsored by [perun](https://github.com/Helmholtz-AI-Energy/perun)\n\n| function                                  |   mpi_ranks | device   | metric   |        value |    ref_value |         std |   % change | type            | alert   |   lower_quantile |   upper_quantile |
|-------------------------------------------|-------------|----------|----------|--------------|--------------|-------------|------------|-----------------|---------|------------------|------------------|
| heat_benchmarks                           |           1 | GPU      | RUNTIME  | 85.3719      | 55.4595      | 0.302896    |    53.9357 | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_robust_scaler_and_inverse   |           1 | GPU      | RUNTIME  | 59.9882      | 30.1147      | 0.13416     |    99.199  | jump-detection  | True    |       nan        |        nan       |
| heat_benchmarks                           |           4 | CPU      | RUNTIME  |  9.87006     |  5.88385     | 0.0207335   |    67.7484 | jump-detection  | True    |       nan        |        nan       |
| matmul_split_0                            |           4 | CPU      | RUNTIME  |  0.128178    |  0.109532    | 0.0320704   |    17.0233 | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_standard_scaler_and_inverse |           4 | CPU      | RUNTIME  |  0.00779852  |  0.00568767  | 0.000951708 |    37.1128 | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_min_max_scaler_and_inverse  |           4 | CPU      | RUNTIME  |  0.00122455  |  0.00100947  | 5.04246e-05 |    21.3066 | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_max_abs_scaler_and_inverse  |           4 | CPU      | RUNTIME  |  0.000800323 |  0.000525618 | 5.13368e-05 |    52.2634 | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_robust_scaler_and_inverse   |           4 | CPU      | RUNTIME  |  4.9826      |  2.43294     | 0.0562545   |   104.798  | jump-detection  | True    |       nan        |        nan       |
| apply_inplace_normalizer                  |           4 | CPU      | RUNTIME  |  0.00116053  |  0.000988138 | 7.57724e-05 |    17.4458 | jump-detection  | True    |       nan        |        nan       |
| qr_split_0                                |           4 | CPU      | RUNTIME  |  0.220444    |  0.985269    | 0.00548881  |   -77.6261 | trend-deviation | True    |         0.221199 |          1.37516 |\n\n[Grafana Dashboard](http://57bc8d92-72f2-4869-accd-435ec06365cb.ka.bw-cloud-instance.org:3000/)\n\nLast updated: 2024-08-06T12:25:54Z

**Review by mtar — APPROVED** [review]
That's nice!
For the public grafana dashboard, we should use a different domain name and use secure connection, such that anyone following the link don't get scared :fearful: .

**Comment by JuanPedroGHM** [comment]
> That's nice! For the public grafana dashboard, we should use a different domain name and use secure connection, such that anyone following the link don't get scared 😨 😉 .

I'll need to find out how to do that one, but it should not be that hard right?

File: benchmarks/cb/manipulations.py [hunk 1]
```diff
@@ -15,6 +16,13 @@ def reshape(arrays):
         a = ht.reshape(array, (10000000, -1), new_split=1)
 
 
+@monitor()
+def resplit(array, new_split: List[int | None]):
```
[the remark this answers is not in the document]
Comment (JuanPedroGHM) [reply benchmarks/cb/manipulations.py:20]: Probably should change it to be compliant to 3.9+. I'm guessing it should be `List[Optional[int]]` for it to be backwards compatible?

[the remark this answers is not in the document]
Comment (mtar) [reply benchmarks/cb/manipulations.py:20]: The the built-in `list` type was already introduced in 3.9.
If you want to have the same version for the benchmarks, then it should be Optional or the older Union type.

**Comment by mtar** [comment]
The documentation has a chapter about setting up HTTPS: https://grafana.com/docs/grafana/latest/setup-grafana/set-up-https/

**Review by mtar — APPROVED** [review]

**Pull request merged**